Refactor development guide (#874)
nsidc/earthaccess: docs/contributing/development.md

   ```bash
   git clone git@github.com:{my-username}/earthaccess
   ```

> 1. We recommend using a feature when developing new features, fixing bugs, updating docs, etc. It's best to give the branch

Reviewer: ```suggestion
1. We recommend creating a feature branch when developing new features, fixing bugs, updating docs, etc. It's best to give the branch
```

    ```bash
    mamba env update -f environment.yml
    mamba activate earthaccess
    ```

    This will update (or create if missing) the `earthaccess` environment and active it. The `earthaccess` package will
    be installed into the environment in editable mode with the optional development dependencies.

    !!! note

        On Sept. 23, 2024, the name of the conda environment changed from `earthaccess-dev` to `earthacess` to align with
        community best practices. If you have an `earthaccess-dev` environment, we recommend deleting it and creating a new one.
        From the repository root, you can do that with these commands:

        ```bash
        mamba env update -f environment.yml
        mamba activate earthaccess
        mamba env remove -n earthaccess-dev
        ```

=== "`pipx`+`nox`"

    `pipx` is a tool to help you install and run end-user applications written in Python and `nox` is Python application
    that automates testing in multiple Python environments. That means, we can let `pipx` and `nox` manage the `earthaccess`
    development environment(s) entirely without having to set one up locally. See the `pipx` installation instructions here:

>     <https://github.com/pypa/pipx?tab=readme-ov-file#install-pipx>.
> 
>     Once you have `pipx` installed, you can either run `nox` without installing it via:

Reviewer: ```suggestion
    Once you have [installed `pipx` following the official instructions](https://github.com/pypa/pipx?tab=readme-ov-file#install-pipx), you can either run `nox` without installing it via:
```

I prefer to avoid bare links in the docs and instead have labeled links.

    ```bash
    pipx run nox [NOX_ARGS]
    ```

>     or install `nox` into an isolated environment and run it with

Reviewer: ```suggestion
    or install `nox` into an isolated environment and run it with:
```

    ```bash
    pipx install nox
    nox [NOX_ARGS]
    ```

>     `nox` handles everything for you, including setting up a temporary virtual environment for each test session.

Reviewer: ```suggestion
    `nox` handles everything for you, including setting up a temporary virtual environment for each task (e.g. running tests, building docs, etc.) you need to run.
```

Now that your development environment is set up, you can run the tests.

## Running tests

> We recommend using `nox` to run the various test "sessions" (tasks) provided by `earthaccess`. To use, run `nox` without

Reviewer: ```suggestion
We recommend using `nox` to run the various "sessions" (`nox`'s term for tasks) provided by `earthaccess`. To use, run `nox` without
```

> any arguments:
> ```bash

Reviewer: ```suggestion
any arguments:

```bash
```

In order to develop new features or fix bugs etc. we need to set up a virtual
environment and install the library locally.

## Development environment setup

There are a few options for setting up a development environment; you can use Python's `venv`, use `conda`/`mamba`, or _not_
manage one yourself and use `pipx` to run tests and build docs with `nox`.

* If you're a Windows user, you'll likely want to set up an environment yourself with `conda`/`mamba`.
* If you're working in a JupyterHub, you'll likely want to set up an environment yourself with `conda`/`mamba`.
* If you're using an IDE like VS Code or PyCharm, you'll likely want to set up an environment yourself with `venv` or `conda`/`mamba`.
* If you're using a plain text editor and don't know how to or want to manage a virtual environment, you'll likely want to start with `pipx`.

> === "`venv`"

Reviewer: Something I really don't like about the tabbed sections as they are now is that it's very difficult to see where the bottom is and regular instructions resume. I pushed a commit subtly changing the CSS and I think it makes a huge difference. What do you think?

## New

![image](https://github.com/user-attachments/assets/c769029f-629f-49c9-9cd4-64a6e44692f8)

## Current

![image](https://github.com/user-attachments/assets/77607ec9-ead3-4202-bd49-abb608af1a3a)

Author: Yes, I like having the outline; I'd be open to making the tabs and box even more obvious -- I might play around with it.

Author: I've updated it to match the tables in our API Docs. 

Now: 
![Uploading image.png…]()

API Docs:
![image](https://github.com/user-attachments/assets/9c52048f-addf-496b-9a0e-ebc0118968a7)

Reviewer: Love it! :star_struck:
